src/activities/quizzes/timing/QuizTimingEntity.js

```diff
@@ -0,0 +1,76 @@
+import { Entity } from '../../../es6/Entity';
+import { Actions, Classes } from '../../../hypermedia-constants';
+import { performSirenAction } from '../../../es6/SirenAction.js';
+
+/**
+ * A QuizTiming subentity of a d2l Quiz.
+ */
+export class QuizTimingEntity extends Entity {
+	/**
+	 * @returns {bool} Whether or not the edit timing type action is present
+	 */
+	canEditTiming() {
+		return this._entity && this._entity.hasActionByName(Actions.quizzes.timing.updateType);
+	}
+
+	timingTypes() {
+		const action = this._entity && this._entity.getActionByName(Actions.quizzes.timing.updateType);
+		if (!action) return;
+		const field = action.getFieldByName('timingType');
+		if (!field) return;
+		return field.value;
+	}
+
+	submissionLateType() {
+		const entity = this._entity && this._entity.getSubEntityByClass(Classes.quizzes.timing.enforced);
+		if (!entity) return;
+		const action = entity.getActionByName(Actions.quizzes.timing.updateLateTypeId);
+		if (!action) return;
+		const field = action.getFieldByName('submissionLateTypeId');
+		if (!field) return;
+		return field.value;
+	}
+	enforcedTimeProperties() {
+		const entity = this._entity && this._entity.getSubEntityByClass(Classes.quizzes.timing.enforced);
+		if (!entity) return;
+		return entity.properties;
```

Comment:
I think it'll be better to have multiple functions to extract specific things out of properties instead of returning all the properties. Similar to how we have specific functions for extracting specific things from actions, rather than returning the whole `actions` array.

Reply:
Good recommendation, I've extracted the specific properties